sql/src/main/java/io/crate/operation/collect/files/FileReadingCollector.java

```diff
@@ -113,7 +114,33 @@ public UriWithGlob(URI uri, URI preGlobUri, Predicate<URI> globPredicate) {
             Predicate<URI> globPredicate = null;
             Matcher hasGlobMatcher = HAS_GLOBS_PATTERN.matcher(uri.toString());
             if (hasGlobMatcher.matches()) {
-                preGlobUri = URI.create(hasGlobMatcher.group(1));
+                if (fileUri.startsWith("/") || fileUri.startsWith("file://")) {
+                    /*
+                     * Substitute a symlink with the real path.
+                     * The wildcard needs to be maintained, though, because it is used to generate the matcher.
+                     * Take the part before the wildcard (*) and try to resolved the real path.
+                     * If the part before the wildcard contains a part of the filename (e.g. /tmp/foo_*.json) then use the
+                     * parent directory of this filename to resolved the real path.
+                     * Then replace this part with the real path and generate the URI.
+                     */
+                    Path oldPath = Paths.get(toURI(hasGlobMatcher.group(1)));
+                    if (!oldPath.toFile().isDirectory()) {
```

Comment:
I think this could be replaced with `Files.isDirectory(oldPath)` in order to avoid the toFile conversion.